File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    const hash = new Set(phone_book);

    for (let phone_number of hash) {
        let prefix = "";
        for (let char of phone_number) {
            prefix += char
            if (hash.has(prefix) && prefix !== phone_number) {
                return false
            }
        }
    }

    return true;
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def isStartWith(text, prefix):
    return text.startswith(prefix)


def solution(phone_book):
    phone_book.sort()
    for i in range(len(phone_book) - 1):
        if isStartWith(phone_book[i + 1], phone_book[i]):
            return False

    return True